src/odemis/gui/cont/milling.py

```
@@ -102,15 +102,15 @@ def _to_physical_position(pos: Tuple[float, float], stream: acqstream.Stream) ->
 
 # TODO: support other shapes
 def rectangle_pattern_to_shape(canvas,
-                        stream: acqstream.Stream,
+                        ref_img: model.DataArray,
                         pattern: RectanglePatternParameters,
                         colour: str = "#FFFF00",
                         name: str = None) -> EditableShape:
```

Ruff (0.15.2)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)